### ISSUE TITLE
Type analyze map and filter

### ISSUE THREAD

**Issue by benjaminsaljooghi** [microsoft/pyright#492]
**Is your feature request related to a problem? Please describe.**
In-built functions `map` and `filter` are not type analyzed by Pyright

**Describe the solution you'd like**
Pyright should check that the input iterable of `map` and `filter` is compatible with the provided function.

**Additional context**
```Python
def add_one(x: int) -> int:
    return x + 1

def is_one(x: int) -> bool:
    return x == 1

nums = ["a", "b", "c"]

ones = filter(is_one, nums) # Pyright should raise an error here.
added = map(add_one, nums) # Pyright should raise an error here.

```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.22, which I just published.
